### PR TITLE
Fix NoSuchMethodError in mvc-annotation-commons&change deprecated method

### PR DESCRIPTION
**Pull request by CzyerChen**
### Fix [NoSuchMethodError in mvc-annotation-commons&change deprecated method](https://github.com/apache/skywalking/issues/11652)
- [x] Add a unit test to verify that the fix works. 
current test scenarios(spring-4.3.x-scenario, spring-6.x-scenario) passed.
- [x] Explain briefly why the bug exists and how to fix it.

- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes https://github.com/apache/skywalking/issues/11652.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
I want to double-check, [this test](https://github.com/apache/skywalking-java/tree/main/test/plugin/scenarios/spring-3.1.x-scenario) has covered 3.2.18.RELEASE, so, only 3.2.0 had the bug, but 3.2.8 is OK?

**Comment by CzyerChen** [comment]
> I want to double-check, [this test](https://github.com/apache/skywalking-java/tree/main/test/plugin/scenarios/spring-3.1.x-scenario) has covered 3.2.18.RELEASE, so, only 3.2.0 had the bug, but 3.2.8 is OK?

I can write a scenario with version `3.2.18.RELEASE` and test again locally.

**Comment by wu-sheng** [comment]
The link I provided, it has tested 3.1.4.RELEASE, 3.2.18.RELEASE, 4.0.9.RELEASE, check this, https://github.com/apache/skywalking-java/blob/main/test/plugin/scenarios/spring-3.1.x-scenario/support-version.list.

What I want to confirm with you is, is this test covers your scenario if we add 3.2.0.xx into that? Or 3.2.18 is still having this bug just not covered by the test.

**Comment by CzyerChen** [comment]
> The link I provided, it has tested 3.1.4.RELEASE, 3.2.18.RELEASE, 4.0.9.RELEASE, check this, https://github.com/apache/skywalking-java/blob/main/test/plugin/scenarios/spring-3.1.x-scenario/support-version.list.
> 
> What I want to confirm with you is, is this test covers your scenario if we add 3.2.0.xx into that? Or 3.2.18 is still having this bug just not covered by the test.

- SpringBoot version upgrade to 3.2.0 and Spring version upgrade to 6.1.1, then `getMethodValue` removed and caused `NoSuchMethodError`, so i want to commit this change.
- I add Spring version 6.1.0/6.1.1 support-version into `spring-6.x-scenario`, change dependency versions in pom.xml to support Spring 6.x([springboot2+spring5 dependency management for 2.7.x with springboot3+spring6 dependency management for 3.0.x ](https://docs.spring.io/spring-boot/docs/3.0.x/reference/html/dependency-versions.html#appendix.dependency-versions).
- Upgrade from springboot3.1.x+spring6.0.x to springboot3.2.x+spring6.1.x, our testcase `spring-6.x-scenario` is a spring project, not a springboot project, so i follow the issue [http request parameter error](https://github.com/spring-projects/spring-boot/issues/38541#issuecomment-1827462871) to change maven compile config, and it works. 
- I passed `spring-6.x-scenario` locally.

File: test/plugin/scenarios/spring-6.x-scenario/support-version.list [hunk 1]
```diff
@@ -15,3 +15,5 @@
 # limitations under the License.
 
 6.0.4
+6.1.0
+6.1.1
```
Comment (wu-sheng) [inline test/plugin/scenarios/spring-6.x-scenario/support-version.list:19]: So, only this is using SpringBoot 3.2.0, which was failing before the fix, am I getting your idea correctly?

Comment (CzyerChen) [reply test/plugin/scenarios/spring-6.x-scenario/support-version.list:19]: yes, I tracd my own springboot project in 3.2.0(Spring 6.1.1), and find SpringMVC node missing, but springboot project in 3.1.6(Spring 6.0.14) works.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**